c7n/filters/kms.py

```diff
@@ -31,6 +30,24 @@ class KmsRelatedFilter(RelatedResourceFilter):
     RelatedResource = "c7n.resources.kms.Key"
     AnnotationKey = "matched-kms-key"
 
+    def get_related(self, resources):
+        resource_manager = self.get_resource_manager()
```

Comment:
could you super() the duplication away?

Reply:
Maybe? I couldn't see a way to do it that was helpful though. Because super() gives us a map of key id -> key, but loses relations based on alias (if we have a queue with the key id showing alias/aws/sqs, for example). So we end up having to pull from resource_manager again to cover the alias case.

(I thought about refactoring `RelatedResourceFilter.get_related()` so that super() would be more useful, but dismissed the idea. The fact that it doesn't work smoothly here felt like a KMS-specific concern more than a general related filter concern. Maybe it's worth rethinking that.)